test/system/src/test/java/io/pravega/test/system/AbstractFailoverTests.java

```diff
@@ -193,14 +188,19 @@ void performFailoverTest() throws InterruptedException {
                     } else {
                         log.debug("Read timeout");
                     }
-                } catch (ConnectionClosedException e) {
+                } catch (RetriesExhaustedException e) {
                     log.warn("Test exception in reading events: ", e);
                     continue;
-                } catch (ReinitializationRequiredException e) {
+                } catch (Throwable e) {
+                    if (e.getCause() instanceof RetriesExhaustedException) {
```

Comment:
I don't understand why this is needed given that we are already catching `RetriesExhaustedException` above.